patches/api/0421-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
++
++    // AxolotlBucket
++    PropertyKey<Axolotl.Variant> VARIANT = PropertyKey.of("Variant", Axolotl.Variant.class);
++
++    // MusicInstrumentMeta
++    PropertyKey<MusicInstrument> INSTRUMENT = PropertyKey.of("instrument", MusicInstrument.class);
++
++    // BannerMeta
++    PropertyKey<DyeColor> SHIELD_BANNER_BASE = PropertyKey.of("Base", DyeColor.class);
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..fd286c088caa098447ae857d341a09ea6a936f45
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+@@ -0,0 +1,26 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.potion.PotionEffectType;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a serialized stew effect with a duration.
++ */
++public interface StewEffect {
++
++    /**
++     * Gets the effect type
++     * @return type
++     */
++    @NotNull PotionEffectType effect();
++
++    /**
++     * Gets the duration override for this
++     * stew effect instance.
++     *
++     * @return duration override
++     */
++    @Nullable
++    Integer duration();
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java b/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..3ae8e35cb98c2b39714f3347ad49a474e505b45a
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java
+@@ -0,0 +1,29 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.DyeColor;
++import org.bukkit.entity.TropicalFish;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Represents a serialized tropical fish.
++ */
++public interface TropicalFishVariant {
++
++    /**
++     * Gets the pattern color
++     * @return color
++     */
++    @NotNull DyeColor patternColor();
++
++    /**
++     * Gets the body color
++     * @return color
++     */
++    @NotNull DyeColor bodyColor();
++
++    /**
++     * Gets the type
++     * @return type
++     */
++    @NotNull TropicalFish.Pattern type();
++}
+diff --git a/src/main/java/io/papermc/paper/property/MutablePropertyView.java b/src/main/java/io/papermc/paper/property/MutablePropertyView.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..d73a2276a3e56905e9595f85f5532f06f8dc583d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/property/MutablePropertyView.java
+@@ -0,0 +1,56 @@
++package io.papermc.paper.property;
++
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++public interface MutablePropertyView extends PropertyView, Cloneable {
```

Comment:
I'm not sure the words `Mutable` and `View` belong together in this context. Doesn't a view usually imply that you can't change it, but the underlying data can change so you are able to read changes?